### PR TITLE
[LibUI - React] Providers, Payment methods and Flags icons in LibUi

### PR DESCRIPTION
**Pull request by mlegall-ledger**
-Added new icons to the lib (Payment, Providers, Country Flags) -New script to build icons from sources (SVG)
-New components in the react lib
-New stories in react storybook

### 📝 Description

-Added new icons to the lib (Payment, Providers, Country Flags)
-New script to build icons from sources (SVG)
-New components in the react lib
-New stories in react storybook

### ❓ Context

- **Impacted projects**:  LibUI
- **Linked resource(s)**: LIVE-3644 LIVE-3645 LIVE-3646

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

https://user-images.githubusercontent.com/39890664/195899618-4601adc4-adfa-419e-a8c8-6533b500caf8.mov

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: libs/ui/packages/icons/scripts/buildAdditionalIcons.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  
+      function ${componentName} ({ size = 16 }: Props): JSX.Element {
+        return ${jsx};
+      }
+  
+      ${exports}
+    `;
+}
+
+console.log('START OF PART 2')
```
Comment (elbywan) [inline libs/ui/packages/icons/scripts/buildAdditionalIcons.js:35]: could you please remove the logs?

File: libs/ui/packages/icons/scripts/buildAdditionalIcons.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
+const fs = require("fs");
```
Comment (elbywan) [inline libs/ui/packages/icons/scripts/buildAdditionalIcons.js:1]: Are you sure you need a separate script? At the very least I see a lot of duplicated code between the two scripts which could be externalized in a separate file imported by both.

File: libs/ui/packages/icons/scripts/buildAdditionalIcons.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+                svg,
+                {...options, template: reactTemplate},
+                {componentName: name},
+                `${reactDir}/${folderName}/${name}.tsx`
+            )
+        })
+    });
+ })
+
+ console.log('END OF PART 2')
```
Comment (mcayuelas-ledger) [inline libs/ui/packages/icons/scripts/buildAdditionalIcons.js:100]: 🔥

Comment (mcayuelas-ledger) [reply libs/ui/packages/icons/scripts/buildAdditionalIcons.js:100]: Didn't see elbywan's comments !

File: libs/ui/packages/icons/scripts/buildAdditionalIcons.js (hunk 2 above)
Comment (mlegall-ledger) [reply libs/ui/packages/icons/scripts/buildAdditionalIcons.js:1]: It's planned to merge those scripts, Initially for this task those icons were in separate packages but we decided to regroup everything in the Icons package. With not much time left I didn't had time to merge the scripts.

File: libs/ui/packages/icons/scripts/buildAdditionalIcons.js (hunk 3 above)
Comment (mlegall-ledger) [reply libs/ui/packages/icons/scripts/buildAdditionalIcons.js]: it's removed ! :)

File: libs/ui/packages/icons/scripts/buildAdditionalIcons.js (hunk 2 above)
Comment (elbywan) [reply libs/ui/packages/icons/scripts/buildAdditionalIcons.js:1]: > It's planned to merge those scripts, Initially for this task those icons were in separate packages but we decided to regroup everything in the Icons package.

Ah I see 🙇 

> With not much time left I didn't had time to merge the scripts.

Not sure I get it, are you planning to merge these scripts then? What do you mean by `not much time left` ?

**Review by elbywan — CHANGES_REQUESTED** [review]
See comment about having 2 scripts for building similar icons instead of 1.

**Comment by gre** [comment]
CI appear to reveal issues in tests

**Review by elbywan — APPROVED** [review]
For the record, apparently this PR needs to be merged asap for some reason to provide the PTX team with these icons. So I'm approving it.

**BUT** I'm expecting a rework of the scripts soon after the merge 😉 - and I do not like shipping unfinished PRs.

**Pull request merged**